### PR TITLE
[MI-3752]: Added the test cases for the mentions in MS Teams sync plugin

### PR DESCRIPTION
**Pull request by arush-vashishtha**
**Summary**
This PR consist of the test cases for the mentions in MS Teams sync:-

- Mention single user.
- Mention Multiple users.
- Mention all user in channel or GM.

### CODE REVIEW

File: data/test-cases/plugins/ms-teams-sync/mentions/Mention_all_users.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+---
+# (Required) Ensure all values are filled up
+name: "Mention all users in channel or GM"
+status: Active
+priority: Normal
+folder: mentions
+authors: "@AayushChaudhary0001"
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/mentions/Mention_all_users.md:7]: update author

File: data/test-cases/plugins/ms-teams-sync/mentions/Mention_all_users.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. In the text area mention "all" in the desired channel or GM on MM.
+2. Click on the send button in MM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/mentions/Mention_all_users.md:38]: combine these steps if possible

File: data/test-cases/plugins/ms-teams-sync/mentions/Mention_all_users.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+3. Navigate to the linked channel or GM in MS teams.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/mentions/Mention_all_users.md:39]:
```suggestion
3. Navigate to the linked channel or GM in MS Teams.
```

File: data/test-cases/plugins/ms-teams-sync/mentions/Mention_all_users.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+**Step 2**
+
+1. In the text area mention the "Channel" or "everyone" in the desired channel or GM respectively on MS teams.
+2. Click on the send button in MS teams.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/mentions/Mention_all_users.md:44]: same

File: data/test-cases/plugins/ms-teams-sync/mentions/Mention_multiple_users.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. In the text area mention multiple users using '@' in channel or GM on MM.
+2. Click on the send button in MM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/mentions/Mention_multiple_users.md:38]: same

**Review by ayusht2810 — COMMENTED** [review]
should we add special mentions as well like `channel`, `here`?

File: data/test-cases/plugins/ms-teams-sync/mentions/Mention_all_users.md [hunk 6]
```diff
@@ -34,16 +34,15 @@ steps_hashed: null
 
 **Step 1**
 
-1. In the text area mention "all" in the desired channel or GM on MM.
-2. Click on the send button in MM.
-3. Navigate to the linked channel or GM in MS teams.
+1. In the text area mention "all"/"channel"/"here" in the linked channel or GM and click on send button in MM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/mentions/Mention_all_users.md:37]:
```suggestion
1. Send `all`/`channel`/`here` mentions in the linked channel or GM in MM.
```

Comment (ayusht2810) [reply data/test-cases/plugins/ms-teams-sync/mentions/Mention_all_users.md:37]: fix everywhere

File: data/test-cases/plugins/ms-teams-sync/mentions/Mention_all_users.md [hunk 7]
```diff
@@ -34,16 +34,15 @@ steps_hashed: null
...
-2. Click on the send button in MS teams.
-3. Navigate to the linked channel or GM in MM.
+1. In the text area mention the "Channel" or "everyone" in the linked channel or GM respectively and click on send button in MS Teams.
+2. Navigate to the linked channel or GM in MM.
 
 **Expected**
 
-The user should be able to view the 'all' mention highlighted in the desired channel or GM on MS teams. After step 2, the user should be able to view the 'all' mention highlighted in the desired channel or GM on MM.
\ No newline at end of file
+The user should be able to view the "channel name" or "everyone" mention highlighted in the linked channel or GM on MS Teams.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/mentions/Mention_all_users.md:47]:
```suggestion
The user should be able to view the `channel` mention or `everyone` mention highlighted in the linked channel or GM, respectively, on MS Teams. 
```

Comment (ayusht2810) [reply data/test-cases/plugins/ms-teams-sync/mentions/Mention_all_users.md:47]: fix everywhere

File: data/test-cases/plugins/ms-teams-sync/mentions/Mention_all_users.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Send the `all`/`channel`/`here` mention in the linked channel or GM in MM.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/mentions/Mention_all_users.md:37]:
```suggestion
1. Mention all the users through 'all' / 'channel' / 'here' in the linked channel or GM in MM.
```

File: data/test-cases/plugins/ms-teams-sync/mentions/Mention_all_users.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+2. Navigate to the linked channel or GM in MS Teams.
+
+**Step 2**
+
+1. Send the `Channel name` or `everyone` mention in the linked channel or GM respectively in MS Teams.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/mentions/Mention_all_users.md:42]: Same as above

File: data/test-cases/plugins/ms-teams-sync/mentions/Mention_multiple_users.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Send mention for multiple users in the linked channel or GM on MM.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/mentions/Mention_multiple_users.md:37]: Make the steps consistent as mentioned above. Refer to the suggestion provided in the previous test case. Fix everywhere.

File: data/test-cases/plugins/ms-teams-sync/mentions/Mention_single_user.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Send mention for a user in the linked channel or DM/GM on MM.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/mentions/Mention_single_user.md:37]: Same

File: data/test-cases/plugins/ms-teams-sync/mentions/Mention_all_users.md [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Mention all the users through `all`/`channel`/`here` in the linked channel or GM on MM.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/mentions/Mention_all_users.md]: please add @ also

**Review by AayushChaudhary0001 — APPROVED** [review]

File: data/test-cases/plugins/ms-teams-sync/mentions/Mention_multiple_users.md [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+
+**Step 1**
+
+1. Mention multiple users in the linked channel or GM on MM.
+2. Navigate to the linked channel or DM/GM in MS Teams.
+
+**Step 2**
+
+1. Mention multiple users in the linked channel or GM on MS Teams.
+2. Navigate to the linked channel or DM/GM in MM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/mentions/Mention_multiple_users.md:43]: same. check everywhere

File: data/test-cases/plugins/ms-teams-sync/mentions/Mention_all_users.md [hunk 14]
```diff
@@ -0,0 +1,48 @@
+---
+# (Required) Ensure all values are filled up
+name: "Mention all users in channel or GM"
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/mentions/Mention_all_users.md:3]:
```suggestion
name: "Mention all users in the linked channel or GM"
```

Comment (ayusht2810) [reply data/test-cases/plugins/ms-teams-sync/mentions/Mention_all_users.md:3]: check everyhwhere

**Pull request merged**